src/ArmTemplates/Extractor/Utilities/DataProcessors/Absctraction/IDataProcessor.cs

```diff
@@ -0,0 +1,19 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors.Absctraction
+{
+    public interface IDataProcessor<TTemplateResource>
```

Comment:
`TTemplateResource` can be restricted with `where ...` expression:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/where-generic-type-constraint

```suggestion
    public interface IDataProcessor<TTemplateResource> where TTemplateResource : ITemplateResource // please check names, I could write it not properly
```

Reply:
Outdated, https://github.com/Azure/azure-api-management-devops-resource-kit/pull/709#discussion_r875681148